src/main/java/gift/service/OrderService.java

```diff
@@ -0,0 +1,75 @@
+package gift.service;
+
+import gift.dto.kakaomessage.SendKakaoMessageRequest;
+import gift.dto.order.CreateOrderCommand;
+import gift.dto.order.OrderDto;
+import gift.entity.Member;
+import gift.entity.Order;
+import gift.entity.ProductOption;
+import gift.exception.BadRequestException;
+import gift.repository.OrderRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+@Service
+public class OrderService {
+
+    private final ProductOptionService productOptionService;
+    private final MemberService memberService;
+    private final OrderRepository orderRepository;
+    private final KakaoMessageService kakaoMessageService;
+
+    public OrderService(ProductOptionService productOptionService,
+            MemberService memberService, OrderRepository orderRepository,
+            KakaoMessageService kakaoMessageService) {
+        this.productOptionService = productOptionService;
+        this.memberService = memberService;
+        this.orderRepository = orderRepository;
+        this.kakaoMessageService = kakaoMessageService;
+    }
+
+    @Transactional
+    public OrderDto createOrder(CreateOrderCommand command) {
+        Member senderMember = memberService
+                .findMemberByIdNotDeleted(command.senderMemberId());
+        Member recieverMember = command.senderMemberId().equals(command.receiverMemberId()) ?
+                senderMember : memberService.findMemberByIdNotDeleted(command.receiverMemberId());
```

Comment:
```suggestion
        Member recieverMember = memberService.findMemberByIdNotDeleted(command.receiverMemberId());
```
sender 와 receiver가 같으면 이미 가져온 sender를 재사용하여 쿼리 한 번을 이득볼 수 있다고 이해했어요! 하지만 쿼리 한 번을 손해보더라도 가독성을 챙기는 건 어떨까요? 의견이 궁금합니다!

Reply:
네 맞습니다, 쿼리의 이득을 위해서 재사용하는 방식을 채택하였습니다.

답변드리기 위해 여러가지 고민하였는데, 결론적으로는 멘토님의 의견과 쿼리이득을 모두 챙기기 위해 memberService를 리팩토링할 수 있겠다는 생각이 들었습니다.
- 불필요한 쿼리를 추가로 수행하는것과, 가독성의 등가교환이 핵심 내용이라고 생각합니다.
    - 개인적인 생각으로는 불필요한 쿼리는 최소화하되, '검증'해야하는 경우라면 쿼리를 아낌없이 실행해도 된다고 생각합니다.
    - 가독성이 떨어진다면 (분기문 작성 등), 각 기능을 구현하는 메소드를 분리하는것이 적절해보일것 같습니다.
- 하지만 불필요한 쿼리를 실행하지 않고서 (DB 쿼리), 영속성 컨텍스트 내에서 1차 캐시를 활용할 수 있다면, **멘토님께서 제시하신 가독성을 챙기면서 쿼리를 이득볼 수 있는 코드를 작성 가능**하겠습니다.
    - 현재 MemberService는 `select * from member where id = {$id} and deletedAt is null` 같이 id 외의 다른 조건문을 달아서 조회하므로, 필수적으로 db쿼리가 필요합니다.
    - 따라서, deletedAt에 대한 검사를 java단에서 수행하도록 하면, 영속성 컨텍스트 내의 1차 캐시에서 캐싱해올 수 있게 되겠습니다.

반영하여 PR드리겠습니다!